modules/apps/29-fee/keeper/migrations_test.go

```diff
@@ -0,0 +1,167 @@
+
+import (
+	sdkmath "cosmossdk.io/math"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	minttypes "github.com/cosmos/cosmos-sdk/x/mint/types"
+
+	"github.com/cosmos/ibc-go/v8/modules/apps/29-fee/keeper"
+	"github.com/cosmos/ibc-go/v8/modules/apps/29-fee/types"
+	channeltypes "github.com/cosmos/ibc-go/v8/modules/core/04-channel/types"
+)
+
+func (suite *KeeperTestSuite) TestLegacyTotal() {
+	fee := types.NewFee(defaultRecvFee, defaultAckFee, defaultTimeoutFee)
+	expLegacyTotal := sdk.NewCoins(sdk.NewCoin(sdk.DefaultBondDenom, sdkmath.NewInt(600)))
+
+	suite.Require().Equal(expLegacyTotal, keeper.LegacyTotal(fee))
+}
+
+func (suite *KeeperTestSuite) TestMigrate1to2() {
+	var (
+		packetID         channeltypes.PacketId
+		moduleAcc        sdk.AccAddress
+		refundAcc        sdk.AccAddress
+		initRefundAccBal sdk.Coins
+		initModuleAccBal sdk.Coins
+		packetFees       []types.PacketFee
+	)
+
+	testCases := []struct {
+		name     string
+		malleate func()
+		assert   func(error)
+	}{
+		{
+			"success: no fees in escrow",
+			func() {},
+			func(err error) {
+				suite.Require().NoError(err)
+				suite.Require().Empty(suite.chainA.GetSimApp().IBCFeeKeeper.GetAllIdentifiedPacketFees(suite.chainA.GetContext()))
+
+				// refund account balance should not change
+				refundAccBal := suite.chainA.GetSimApp().BankKeeper.GetBalance(suite.chainA.GetContext(), refundAcc, sdk.DefaultBondDenom)
+				suite.Require().Equal(initRefundAccBal[0], refundAccBal)
+
+				// module account balance should not change
+				moduleAccBal := suite.chainA.GetSimApp().BankKeeper.GetBalance(suite.chainA.GetContext(), moduleAcc, sdk.DefaultBondDenom)
+				suite.Require().True(moduleAccBal.IsZero())
+			},
+		},
+		{
+			"success: one fee in escrow",
+			func() {
+				fee := types.NewFee(defaultRecvFee, defaultAckFee, defaultTimeoutFee)
+				packetFee := types.NewPacketFee(fee, refundAcc.String(), []string(nil))
+				packetFees = []types.PacketFee{packetFee}
+			},
+			func(err error) {
+				suite.Require().NoError(err)
+
+				// ensure that the packet fees are unmodified
+				expPacketFees := []types.IdentifiedPacketFees{
+					types.NewIdentifiedPacketFees(packetID, packetFees),
+				}
+				suite.Require().Equal(expPacketFees, suite.chainA.GetSimApp().IBCFeeKeeper.GetAllIdentifiedPacketFees(suite.chainA.GetContext()))
+
+				unusedFee := sdk.NewCoin(sdk.DefaultBondDenom, sdkmath.NewInt(300))
+				// refund account balance should increase
+				refundAccBal := suite.chainA.GetSimApp().BankKeeper.GetBalance(suite.chainA.GetContext(), refundAcc, sdk.DefaultBondDenom)
+				suite.Require().Equal(initRefundAccBal.Add(unusedFee)[0], refundAccBal)
+
+				// module account balance should decrease
+				moduleAccBal := suite.chainA.GetSimApp().BankKeeper.GetBalance(suite.chainA.GetContext(), moduleAcc, sdk.DefaultBondDenom)
+				suite.Require().Equal(initModuleAccBal.Sub(unusedFee)[0], moduleAccBal)
+			},
+		},
+		{
+			"success: many fees with multiple denoms in escrow",
```

Comment:
I think it'd be nice to have a test case for different incentivized packets. If I understand correctly, this is does only a single iteration. If I have the following diffs (removing iteration), the tests pass:
```diff
diff --git a/modules/apps/29-fee/keeper/migrations.go b/modules/apps/29-fee/keeper/migrations.go
index 3eae34f7e..5d5701e8f 100644
--- a/modules/apps/29-fee/keeper/migrations.go
+++ b/modules/apps/29-fee/keeper/migrations.go
@@ -27,19 +27,17 @@ func (m Migrator) Migrate1to2(ctx sdk.Context) error {
        iterator := storetypes.KVStorePrefixIterator(store, []byte(types.FeesInEscrowPrefix))
        defer sdk.LogDeferred(ctx.Logger(), func() error { return iterator.Close() })
 
-       for ; iterator.Valid(); iterator.Next() {
-               feesInEscrow := m.keeper.MustUnmarshalFees(iterator.Value())
+       feesInEscrow := m.keeper.MustUnmarshalFees(iterator.Value())
 
-               for _, packetFee := range feesInEscrow.PacketFees {
-                       refundCoins := legacyTotal(packetFee.Fee).Sub(packetFee.Fee.Total()...)
+       for _, packetFee := range feesInEscrow.PacketFees {
+               refundCoins := legacyTotal(packetFee.Fee).Sub(packetFee.Fee.Total()...)
 
-                       refundAddr, err := sdk.AccAddressFromBech32(packetFee.RefundAddress)
-                       if err != nil {
-                               return err
-                       }
-
-                       m.keeper.distributeFee(ctx, refundAddr, refundAddr, refundCoins)
+               refundAddr, err := sdk.AccAddressFromBech32(packetFee.RefundAddress)
+               if err != nil {
+                       return err
                }
+
+               m.keeper.distributeFee(ctx, refundAddr, refundAddr, refundCoins)
        }
```